watch.go

```diff
@@ -0,0 +1,70 @@
+package kubeadmtokentpr
+
+import (
+	"fmt"
+	"time"
+
+	microerror "github.com/giantswarm/microkit/error"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/pkg/api"
+	"k8s.io/client-go/pkg/api/v1"
+	"k8s.io/client-go/tools/cache"
+)
+
+const (
+	WatchTimeout = 90 * time.Second
+)
+
+func SearchToken(k8sClient kubernetes.Interface, clusterID string) (string, error) {
```

Comment:
`GetToken` maybe? When I see `SearchToken` I'm thinking about searching for something inside the token.